Unable to mount secret, no such file or directory.

When attempting to use the vault-ethereum plugin in a docker environment using [the official vault docker image](https://hub.docker.com/_/vault/) I am unable to mount the new plugin and receive a no such file or directory error.

## Detailed Description

Here is the vault service in my docker compose file:

```
vault:
    build:
      context: ./
      dockerfile: vault.dockerfile
    ports:
      - "9200:8200"
    volumes:
      - ../vault/config:/vault/config
      - ../vault/server/data:/vault/data
      - ../vault/logs:/vault/logs
    environment:
      - VAULT_ADDR=http://192.168.1.60:9200
      - VAULT_LOCAL_CONFIG={"api_addr":"http://127.0.0.1:8200","plugin_directory":"/vault/plugins","backend":{"consul":{"scheme":"http", "address":"192.168.1.60:9500","advertise_addr":"http://192.168.1.60", "path":"vault/"}},"listener":{"tcp":{"address":"0.0.0.0:8200","tls_disable":"true"}}}
    command: server
```
I extend the official docker image to attempt to install the vault ethereum plugin:

```
FROM vault:latest

# Make new directory for plugins
RUN mkdir /vault/plugins && \

# Download binary for Ethereum plugin
RUN wget https://github.com/immutability-io/vault-ethereum/releases/download/v0.0.3/vault-ethereum_0.0.3_linux_amd64.zip && \
    unzip ./vault-ethereum_0.0.3_linux_amd64.zip && \
    rm ./vault-ethereum_0.0.3_linux_amd64.zip && \
    mv ./vault-ethereum /vault/plugins/vault-ethereum && \
    chown -R vault:vault /vault/plugins
```

I then perform the following operations (this is a fresh install of vault with **consul** as the backend).

```
docker exec -it development_vault_1 vault init
docker exec -it development_vault_1 vault unseal
....
docker exec -it development_vault_1 vault login token=...
```
```
docker exec -it development_vault_1 sha256sum /vault/plugins/vault-ethereum
Output:
fc4734c431ae0333fad4aa480f539aa8b9b4ddcfec89907c3577a8c3de70399e  /vault/plugins/vault-ethereum
```
```
vault write sys/plugins/catalog/ethereum-plugin sha_256="fc4734c431ae0333fad4aa480f539aa8b9b4ddcfec89907c3577a8c3de70399e" command="vault-ethereum"
Output:
Success! Data written to: sys/plugins/catalog/ethereum-plugin
```
```
docker exec -it development_vault_1 vault secrets enable -path=ethereum -plugin-name=ethereum-plugin plugin

Error:
Error enabling: Error making API request.

URL: POST http://192.168.1.60:9200/v1/sys/mounts/ethereum
Code: 400. Errors:

* fork/exec /vault/plugins/vault-ethereum: no such file or directory

```

ls -lrt /vault/ output
```
drwxr-xr-x    2 root     root             0 Feb 17 12:19 logs
drwxr-xr-x    2 root     root             0 Feb 17 12:19 data
drwxr-xr-x    2 root     root             0 Feb 19 19:15 config
drwxr-xr-x    1 vault    vault         4096 Feb 20 09:05 plugins
drwxr-xr-x    2 vault    vault         4096 Feb 20 09:09 file
```

ls -lrt /vault/plugins output
```
-rwxr-xr-x    1 root     root      19384144 Feb 20 09:05 vault-ethereum
```

## Your Environment

Host OS: Windows 10
Docker: 17.12.0-ce-win47 stable
Vault Docker: 0.9.3
Vault Ethereum: v0.0.3

@zambien this makes the build fail with:

```
go build github.com/ethereum/go-ethereum/crypto/secp256k1: build constraints exclude all Go files in /go/pkg/mod/github.com/ethereum/go-ethereum@v1.8.27/crypto/secp256k1
# github.com/ethereum/go-ethereum/rpc
/go/pkg/mod/github.com/ethereum/go-ethereum@v1.8.27/rpc/endpoints.go:96:19: undefined: ipcListen
/go/pkg/mod/github.com/ethereum/go-ethereum@v1.8.27/rpc/ipc.go:50:10: undefined: newIPCConnection
```

--- a/.gitignore
+++ b/.gitignore
@@ -11,4 +11,4 @@ scripts/**
 **/helper/*.txt
 **/helper/SHA256SUM
 **/vault-ethereum
-.dist
+dist

--- a/.gitignore
+++ b/.gitignore
@@ -11,3 +11,4 @@ scripts/**
 **/helper/*.txt
 **/helper/SHA256SUM
 **/vault-ethereum
+.dist

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,15 @@
+FROM vault:latest as build
+RUN apk add --update alpine-sdk
+RUN apk update && apk add go git
+WORKDIR /app
+ENV GOPATH /app
+ADD . /app/src
+RUN GO111MODULE=on go get github.com/immutability-io/vault-ethereum
+RUN GO111MODULE=on CGO_ENABLED=1 GOOS=linux go install -a github.com/immutability-io/vault-ethereum
+
+FROM vault:latest
+WORKDIR /app
+RUN cd /app
+COPY --from=build /app/bin/vault-ethereum /app/bin/vault-ethereum
+# Prove the binary is now an executable
+CMD /app/bin/vault-ethereum

--- a/README.md
+++ b/README.md
@@ -1,5 +1,6 @@
 ![Immutability](/docs/tagline.png?raw=true "Changes Everything")
 
+
 Ethereum plugin for Vault
 -----------------
 
@@ -18,7 +19,11 @@ To put it another way:  If your org is dealing in crypto you need this or someth
 
 ## No Warrantees Implied
 
-Use of this plugin with real ETH on the mainnet is at your own risk and no warranties should be implied. The guide here describes how to install and run this plugin on a Mac laptop. This plugin can be run on any platform that Vault supports; but, each environment has its own nuances, and for clarity's sake I will only discuss the Mac laptop use case. Running Vault in a production environment in an enterprise requires planning and operational skills. If you would like help running Vault in production, please reach out to [Immutability, LLC](mailto:jeff@immutability.io).
+Use of this plugin with real ETH on the mainnet is at your own risk and no warranties should be implied. The guide here describes how to install and run this plugin on a Mac laptop. This plugin can be run on any platform that Vault supports; but, each environment has its own nuances, and for clarity's sake I will only discuss the Mac laptop use case. Running Vault in a production environment in an enterprise requires planning and operational skills. If you would like help running Vault in production, please reach out to [Immutability, LLC](docker run --rm --privileged \
+-v $(pwd):/go/src/github.com/immutability-io/vault-ethereum \
+-v /var/run/docker.sock:/var/run/docker.sock \
+-w /go/src/github.com/immutability-io/vault-ethereum \
+mailchain/goreleaser-xcgo goreleaser --snapshot --rm-distto:jeff@immutability.io).
 
 ## API
 
@@ -655,6 +660,18 @@ transaction_hash    0x2ff5dd013e5a4d00cf007a7fb689c4ebf50541c2e7ddfaf16212e7ed1b
 ## More Use Cases
 
 There are many more uses cases to be explored; but, I will leave that to you. If you brave enough...
+
+## Releasing
+
+We use goreleaser for our releases.  If you wish to run goreleaser locally, the following will give you a snapshot build for darwin, linux, and windows.
+
+```bash
+docker run --rm --privileged \
+-v $(pwd):/go/src/github.com/immutability-io/vault-ethereum \
+-v /var/run/docker.sock:/var/run/docker.sock \
+-w /go/src/github.com/immutability-io/vault-ethereum \
+mailchain/goreleaser-xcgo goreleaser --snapshot --rm-dist
+```
 
 ## Credits
 

--- a/README.md
+++ b/README.md
@@ -19,11 +19,7 @@ To put it another way:  If your org is dealing in crypto you need this or someth
 
 ## No Warrantees Implied
 
-Use of this plugin with real ETH on the mainnet is at your own risk and no warranties should be implied. The guide here describes how to install and run this plugin on a Mac laptop. This plugin can be run on any platform that Vault supports; but, each environment has its own nuances, and for clarity's sake I will only discuss the Mac laptop use case. Running Vault in a production environment in an enterprise requires planning and operational skills. If you would like help running Vault in production, please reach out to [Immutability, LLC](docker run --rm --privileged \
--v $(pwd):/go/src/github.com/immutability-io/vault-ethereum \
--v /var/run/docker.sock:/var/run/docker.sock \
--w /go/src/github.com/immutability-io/vault-ethereum \
-mailchain/goreleaser-xcgo goreleaser --snapshot --rm-distto:jeff@immutability.io).
+Use of this plugin with real ETH on the mainnet is at your own risk and no warranties should be implied. The guide here describes how to install and run this plugin on a Mac laptop. This plugin can be run on any platform that Vault supports; but, each environment has its own nuances, and for clarity's sake I will only discuss the Mac laptop use case. Running Vault in a production environment in an enterprise requires planning and operational skills. If you would like help running Vault in production, please reach out to [Immutability, LLC](mailto:jeff@immutability.io).
 
 ## API
 
@@ -663,7 +659,25 @@ There are many more uses cases to be explored; but, I will leave that to you. If
 
 ## Releasing
 
-We use goreleaser for our releases.  If you wish to run goreleaser locally, the following will give you a snapshot build for darwin, linux, and windows.
+## Releasing
+
+We use goreleaser to release cross-compiled versions.  The steps to release are:
+
+```
+export GITHUB_TOKEN="A_GITHUB_TOKEN_THAT_HAS_CORRECT_ACCESS_ENTITLEMENTS"
+VERSION="0.0.30"
+rm -rf dist
+git tag -a $VERSION -m "Some description of the release"
+docker run --rm --privileged \
+-v $(pwd):/go/src/github.com/immutability-io/vault-ethereum \
+-v /var/run/docker.sock:/var/run/docker.sock \
+-w /go/src/github.com/immutability-io/vault-ethereum \
+mailchain/goreleaser-xcgo goreleaser --rm-dist
+sha256sum dist/*.zip > dist/SHA256SUMS
+gpg --detach-sign dist/SHA256SUMS
+```
+
+If you wish to run goreleaser locally, the following will give you a snapshot build for darwin, linux, and windows.
 
 ```bash
 docker run --rm --privileged \
@@ -672,6 +686,10 @@ docker run --rm --privileged \
 -w /go/src/github.com/immutability-io/vault-ethereum \
 mailchain/goreleaser-xcgo goreleaser --snapshot --rm-dist
 ```
+
+You can read more about goreleaser here:
+
+https://goreleaser.com/
 
 ## Credits
 

--- a/README.md
+++ b/README.md
@@ -659,8 +659,6 @@ There are many more uses cases to be explored; but, I will leave that to you. If
 
 ## Releasing
 
-## Releasing
-
 We use goreleaser to release cross-compiled versions.  The steps to release are:
 
 ```

--- a/goreleaser.yml
+++ b/goreleaser.yml
@@ -35,10 +35,10 @@ builds:
   - windows
   goarch:
   - amd64
-
 archives:
 - format: zip
   name_template: "{{.ProjectName}}_{{.Version}}_{{.Os}}_{{.Arch}}"
   files:
   - README.md
   - LICENSE
+

--- a/goreleaser.yml
+++ b/goreleaser.yml
@@ -1,0 +1,44 @@
+env:
+- GO111MODULE=on
+before:
+  hooks:
+  - go mod download
+builds:
+- id: vault-ethereum-darwin-amd64
+  ldflags: -s -w -X github.com/immutability-io/vault-ethereum.Version={{.Version}} -X github.com/immutability-io/vault-ethereum.BuildDate={{.Date}} -X github.com/immutability-io/vault-ethereum.CommitHash={{ .ShortCommit }}
+  binary: vault-ethereum_{{.Version}}
+  env:
+  - CGO_ENABLED=1
+  - CC=o64-clang
+  - CXX=o64-clang++
+  goos:
+  - darwin
+  goarch:
+  - amd64
+- id: vault-ethereum-linux-amd64
+  ldflags: -s -w -X github.com/immutability-io/vault-ethereum/pkg/version.Version={{.Version}} -X github.com/immutability-io/vault-ethereum/pkg/version.BuildDate={{.Date}} -X github.com/immutability-io/vault-ethereum/pkg/version.CommitHash={{ .ShortCommit }}
+  binary: vault-ethereum_{{.Version}}
+  env:
+  - CGO_ENABLED=1
+  goos:
+  - linux
+  goarch:
+  - amd64
+- id: vault-ethereum-windows-amd64
+  ldflags: -s -w -X github.com/immutability-io/vault-ethereum/pkg/version.Version={{.Version}} -X github.com/immutability-io/vault-ethereum/pkg/version.BuildDate={{.Date}} -X github.com/immutability-io/vault-ethereum/pkg/version.CommitHash={{ .ShortCommit }}
+  binary: vault-ethereum_{{.Version}}
+  env:
+  - CGO_ENABLED=1
+  - CC=x86_64-w64-mingw32-gcc
+  - CXX=x86_64-w64-mingw32-g++
+  goos:
+  - windows
+  goarch:
+  - amd64
+
+archives:
+- format: zip
+  name_template: "{{.ProjectName}}_{{.Version}}_{{.Os}}_{{.Arch}}"
+  files:
+  - README.md
+  - LICENSE

